x-pack/filebeat/input/s3/input.go

```diff
@@ -667,3 +674,22 @@ func (c *s3Context) Inc() {
 	defer c.mux.Unlock()
 	c.refs++
 }
+
+// isStreamGzipped determines whether the given stream of bytes (encapsulated in a buffered reader)
+// represents gzipped content or not. A buffered reader is used so the function can peek into the byte
+// stream without consuming it. This makes it convenient for code executed after this function call
+// to consume the stream if it wants.
+func isStreamGzipped(r *bufio.Reader) (bool, error) {
+	// Why 512? See https://godoc.org/net/http#DetectContentType
+	buf, err := r.Peek(512)
+	if err != nil && err != io.EOF {
```

Comment:
Nit. Do we have a test case that covers reading an empty object (or smaller than 512 bytes)? If not, could you please add one?

Reply:
Added test case for empty byte slice.